woocommerce/woocommerce-ios
# Fix: Users Unable to Upload Product Images if the store is private

Closes: #12646

## Description
The PR adds a notice in Product detail when a WP.com site is in Private mode since this makes the images not accessible.
This happens just on WP.com sites, since on self-hosted store it's not possible to put a store in the "private" state.
This has been already fixed here https://github.com/woocommerce/woocommerce-android/pull/11383 and here https://github.com/woocommerce/woocommerce-android/pull/11493 on Android.

## Detail implementation
I decided to use a custom table view cell inside the image section instead of showing it as a banner, because it can happen to have another banner already displayed while the user is editing a product (or it can appear later). In this way, this notice in product images is always displayed when the store is private.

## Testing instructions

### Test 1
1. Log in to a *WP.com site* that is `Public`
2. Go to Products -> Tap on some product
3. Verify the images are available (either existing images or button to add one)
4. In `wp-admin` set the site to `Private` mode in Settings ([more info here](https://wordpress.com/support/privacy-settings/))
5. In the app, switch to a different site and switch back (or re-login, or close and reopen the app)
6. Go to Products -> Tap on some product
7. Verify a notice is displayed instead of an image gallery
8. Tap on the notice
9. Verify the privacy settings page is opened
10. Go back
11. In `wp-admin` set the site to `Coming soon` mode in the Settings
12. In the app, switch to a different site and switch back (or re-login, or close and reopen the app)
13. Go to Products -> Tap on some product
14. Verify the images are available (either existing images or button to add one)

### Test 2
1. Create a self-hosted site with WooCommerce but without installing Jetpack or connecting a WP.com user using JCP
2. Log in to the app
3. Start creating a product
4. Verify the image uploading is available

## Screenshots
| Light mode             |  Dark mode |
:-------------------------:|:-------------------------:
![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-23 at 18 34 58](https://github.com/woocommerce/woocommerce-ios/assets/495617/dfb2ff40-39d3-4a82-bb6d-800c29ef7f80) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-23 at 18 35 11](https://github.com/woocommerce/woocommerce-ios/assets/495617/7389e2fc-65f3-4da6-beb5-3ef6f93586e4)



---
